torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -2720,8 +2723,13 @@ c10::intrusive_ptr<Work> ProcessGroupNCCL::pointToPoint(
 
   if (coalescing_state_ & CoalActive) {
     coalescing_state_ |= CoalP2P;
-    coalescedDevices_.push_back(device);
-    coalescedComms_.push_back(ncclComm);
+    if (coalescedDevice_.index() < 0) {
+      coalescedDevice_ = device;
+    } else {
+      TORCH_CHECK(
+          coalescedDevice_.index() == device.index(), MULTI_DEVICE_ERROR_MSG);
+    }
+    coalescedComm_ = ncclComm;
```

Comment:
The device should always stay the same. PyTorch is functioning on "one device per process" mode since #119421. 
The comm is more complicated here. I need to figure out the way of handling multiple PGs.